crypto/sm2/sm2_pmeth.c

```diff
@@ -317,10 +329,10 @@ static const EVP_PKEY_METHOD sm2_pkey_meth = {
     pkey_sm2_cleanup,
 
     0,
-    0,
+    pkey_sm2_paramgen,
 
     0,
-    0,
+    pkey_sm2_keygen,
```

Comment:
So, if I understand this correct previously we could not gen params and keys directly for SM2. You had to gen them for EC and then convert the resulting key to SM2. With these changes we will be able to generate SM2 keys directly (but only on the SM2 curve). The old way should still work, and would allow SM2 over arbitrary curves. Right?

Reply:
Correct.  `test/ecdsatest.c` does a lot of tests of exactly this.  As a matter of fact, you can also generate a SM2 key and then switch its type to `EVP_PKEY_EC`